src/internet_identity/src/storage.rs

```diff
@@ -279,27 +282,19 @@ impl<M: Memory + Clone> Storage<M> {
 
     /// Writes the data of the specified anchor to stable memory.
     pub fn write(&mut self, anchor_number: AnchorNumber, data: Anchor) -> Result<(), StorageError> {
-        let record_number = self.anchor_number_to_record(anchor_number)?;
-        let buf = candid::encode_one(data).map_err(StorageError::SerializationError)?;
-        self.write_entry_bytes(record_number, buf)
-    }
-
-    fn write_entry_bytes(&mut self, record_number: u32, buf: Vec<u8>) -> Result<(), StorageError> {
-        if buf.len() > self.candid_entry_size_limit() {
+        let storable_anchor = StorableAnchor::from(data);
+        let buf = storable_anchor.to_bytes();
+        if buf.len() > self.header.entry_size as usize {
             return Err(StorageError::EntrySizeLimitExceeded {
                 space_required: buf.len() as u64,
-                space_available: self.candid_entry_size_limit() as u64,
+                space_available: self.header.entry_size as u64,
             });
         }
 
+        let record_number = self.anchor_number_to_record(anchor_number)?;
         let address = self.record_address(record_number);
-        let mut writer = BufferedWriter::new(
-            self.header.entry_size as usize,
-            Writer::new(&mut self.anchor_memory, address),
-        );
-        writer
-            .write_all(&(buf.len() as u16).to_le_bytes())
-            .expect("memory write failed");
+        let mut writer = Writer::new(&mut self.anchor_memory, address);
```

Comment:
We no longer need buffered reader / writers to interact with stable memory, as we write all of the parts (length and data) in one go.